common-model/src/main/java/com/sequenceiq/common/model/EndpointType.java

```diff
@@ -0,0 +1,29 @@
+package com.sequenceiq.common.model;
+
+public enum EndpointType {
+    /*
+    AWS:    do not use private endpoints
+    AZURE:  invalid
+     */
+    NONE,
+
+    /*
+    AWS:    use private endpoints
+    AZURE:  use service endpoints
+     */
+    USE_SERVICE_ENDPOINT,
+
+    /*
+    AWS:    invalid
```

Comment:
shouldn't we map this to AWS private endpoints, too?

Reply:
I would not do it within the scope of this PR, I think this needs more thinking, together with validation for ServiceEndpointCreation. I will disable validation for ServiceEndpointCreation as well.